# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsStyleTransformMatrix.cpp
# Commit: 473ebe5ec7aa
# Full Hash: 473ebe5ec7aa4f1c5da45ed580fb56258737a210
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-08-23 04:21:33
# Regressor Bug: 1819464
# File Overlap Count: 2
# Description:
#   Bug 1819464 - Part 2: Add content-box and stroke-box into transform-box in style. r=emilio
#   
#   Also, add a prefernce for content-box and stroke-box. So we have to
#   define a new type, TransformBox, and let content-box and stroke-box
#   behind the preference.
# ==============================================================================

diff -r 9c22e16e0567 -r 473ebe5ec7aa layout/style/nsStyleTransformMatrix.cpp
--- a/layout/style/nsStyleTransformMatrix.cpp	Tue Aug 22 19:49:03 2023 +0000
+++ b/layout/style/nsStyleTransformMatrix.cpp	Tue Aug 22 19:49:03 2023 +0000
@@ -45,10 +45,17 @@
 
   MOZ_ASSERT(mFrame);
 
+  const auto box = mFrame->StyleDisplay()->mTransformBox;
+  if (box == StyleTransformBox::ContentBox ||
+      box == StyleTransformBox::StrokeBox) {
+    // TODO: Implement this in the following patches.
+    return;
+  }
+
   mIsCached = true;
 
   if (mFrame->HasAnyStateBits(NS_FRAME_SVG_LAYOUT)) {
-    if (mFrame->StyleDisplay()->mTransformBox == StyleGeometryBox::FillBox) {
+    if (box == StyleTransformBox::FillBox) {
       // Percentages in transforms resolve against the SVG bbox, and the
       // transform is relative to the top-left of the SVG bbox.
       nsRect bboxInAppUnits = nsLayoutUtils::ComputeGeometryBox(
@@ -63,11 +70,9 @@
       mHeight = bboxInAppUnits.height;
     } else {
       // The value 'border-box' is treated as 'view-box' for SVG content.
-      MOZ_ASSERT(
-          mFrame->StyleDisplay()->mTransformBox == StyleGeometryBox::ViewBox ||
-              mFrame->StyleDisplay()->mTransformBox ==
-                  StyleGeometryBox::BorderBox,
-          "Unexpected value for 'transform-box'");
+      MOZ_ASSERT(box == StyleTransformBox::ViewBox ||
+                     box == StyleTransformBox::BorderBox,
+                 "Unexpected value for 'transform-box'");
       // Percentages in transforms resolve against the width/height of the
       // nearest viewport (or its viewBox if one is applied), and the
       // transform is relative to {0,0} in current user space.